src/runtime_src/core/edge/user/aie/aie.cpp

```diff
@@ -113,25 +119,31 @@ open_context(const xrt_core::device* device, xrt::aie::access_mode am)
 {
 #ifndef __AIESIM__
   auto drv = ZYNQ::shim::handleCheck(device->get_device_handle());
+  std::lock_guard access_lock_guard(am_lock);
+
+  /* If AIE context is already opened then return from here */
+  if (access_mode != xrt::aie::access_mode::none)
+	  return;
 
   auto current_am = drv->getAIEAccessMode();
   if (current_am != xrt::aie::access_mode::none)
     throw xrt_core::error(-EBUSY, "Can not change current AIE access mode");
 
   int ret = drv->openAIEContext(am);
   if (ret)
-      throw xrt_core::error(ret, "Fail to open AIE context");
+    throw xrt_core::error(ret, "Fail to open AIE context");
 
   drv->setAIEAccessMode(am);
   access_mode = am;
 #endif
 }
 
-bool
+xrt::aie::access_mode
 Aie::
-is_context_set()
+get_access_mode()
 {
-  return (access_mode != xrt::aie::access_mode::none);
+  std::lock_guard access_lock_guard(am_lock);
+  return access_mode;
```

Comment:
I am unsure this solves always the problem.
The access to `access_mode` is safe but is it possible to have another thread changing it after `get_access_mode()` returns before taking an action?
There should be a design top-level comment explaining the precondition and invariant of the code.